rp2040-hal/src/spi.rs

```diff
@@ -45,10 +46,29 @@ impl State for Disabled {}
 impl State for Enabled {}
 
 /// Pac SPI device
-pub trait SpiDevice: Deref<Target = pac::spi0::RegisterBlock> + SubsystemReset {}
+pub trait SpiDevice: Deref<Target = pac::spi0::RegisterBlock> + SubsystemReset {
+    /// The DREQ number for which TX DMA requests are triggered.
+    fn tx_dreq() -> u8;
+    /// The DREQ number for which RX DMA requests are triggered.
+    fn rx_dreq() -> u8;
+}
 
-impl SpiDevice for pac::SPI0 {}
-impl SpiDevice for pac::SPI1 {}
+impl SpiDevice for pac::SPI0 {
```

Comment:
One could use https://docs.rs/rp2040-pac/latest/rp2040_pac/dma/ch/ch_ctrl_trig/enum.TREQ_SEL_A.html here, instead of "magic numbers"

Reply:
Resolved in [243af46](https://github.com/rp-rs/rp-hal/pull/457/commits/243af465d2e2715ca2a3167b7943e8a2a83d85d2)